### ISSUE TITLE
Libnd4j: Add temporal convolution support (causal conv1d op)

### ISSUE THREAD

**Issue by AlexDBlack** [eclipse/deeplearning4j#7629]
See section 3.2 here: https://arxiv.org/pdf/1803.01271.pdf

Idea is very simple: it's basically standard conv1d with 2 small changes:
* Amount of padding so that output size is same as input size
* Causal convolutions

Causal 1d convolutions are standard 1d convolutions but shifted so that their receptive field is based on current and past time steps, not future time steps. Same math, just offset receptive field.

![image](https://user-images.githubusercontent.com/2360237/56721641-a8664a80-6788-11e9-9d0f-a5b0a96ea146.png)

**Comment by sercharpak** [comment]
Some (maybe useful) links:
The original author's [pytorch implementation](https://github.com/locuslab/TCN)
Philippe Remy's [keras implementation](https://github.com/philipperemy/keras-tcn)
And the very recent [stochastic version](https://ait.ethz.ch/projects/2019/stcn/)

**Comment by zhangy10** [comment]
Hey @AlexDBlack, I am just wondering whether this TCN implementation is added to the released version like beta4 or beta5? If in beta5, should I just type the TCN and import it? Thanks for that.

**Comment by AlexDBlack** [comment]
Causal Conv1d support was added here, for both DL4J and DL4J Keras import: https://github.com/KonduitAI/deeplearning4j/pull/107

It will be merged back to eclipse/deeplearning4j soon, and will be in the upcoming 1.0.0-beta6 release.
